### PR TITLE
Fill in the addTodo function 

### PR DESCRIPTION
**Pull request by Karyum**
So we added lines for addTodo function to make it work and a test which we tested and passed.

### CODE REVIEW

File: Test/test.js [hunk 1]
```diff
@@ -1 +1,24 @@
...
+  assert.deepEqual(
+    addTodo(todos, newTodo),
+    [{ description: 'td1' }, { id:1, description: 'td2',  }],
+    "added the second to do"
+  );
+
+  assert.deepEqual(
+    addTodo(todos, newTodo),
+    [{ description: 'td1' }, { id:2, description: 'td2',  }],
+    "added the third to do"
```
Comment (mattlub) [inline Test/test.js]: All three of your 'assert.deepEqual' are the same test!!!

**Review by mattlub — CHANGES_REQUESTED** [review]

File: todo.js [hunk 2]
```diff
@@ -21,6 +21,9 @@ var todo = (function() {
       }
     })(),
     addTodo: function (todos, newTodo) {
+      newTodo.id = todoFunctions.generateId();
+        arr = todos.concat(newTodo);
```
Comment (RQ16) [inline todo.js]: arr is not defined with var.

File: Test/test.js [hunk 3]
```diff
@@ -1 +1,47 @@
...
+    addTodo(todos, newTodo),
+    [{ id:1, description: 'td2'}],
+    "added the second to do"
+  );
+
+  assert.deepEqual(
+    addTodo(todos, newTodo),
+    [{id:2, description:'td2'}],
+    "added the third to do"
+  );
```
Comment (mattlub) [inline Test/test.js]: your tests are still doing the same thing. One thing you could do instead is add two new todos and check that they have different id's.

**Review by mattlub — CHANGES_REQUESTED** [review]

**Review by RQ16 — CHANGES_REQUESTED** [review]

**Comment by mattlub** [comment]
👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍

**Review by RQ16 — APPROVED** [review]

**Review by mattlub — APPROVED** [review]

**Pull request merged**